### PR TITLE
Student profile update added

### LINKED ISSUE #254: API: Update student profile
Create an endpoint POST `/api/profile/student` that will update the following information from the body:
- name
- timeZone

To get the user's id, pass `req.user` from router to controller and use `req.user.id`.

Bonus points:
- validate that user exists and return 404 if it doesn't
- validate that name is not an empty string and return 400 if it is
- validate that the user is a student and return 404 if it isn't.

See instructions on the authentication in #255.

### PR DESCRIPTION
**Pull request by Elia-Git**
# Description

Please provide a short summary explaining what this PR is about.

Fixes #254 

# How to test?

Please provide a short summary of how your changes can be tested?

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/server/api/routes/profile.router.js [hunk 1]
```diff
@@ -30,6 +31,40 @@ router.get('/student', (req, res, next) => {
     .then((result) => res.json(result))
     .catch(next);
 });
+/**
+ * @swagger
+ * /profile/student:
+ *  patch:
+ *    summary: Update all students profile
```
Comment (tpetrina) [inline src/server/api/routes/profile.router.js]: What do you mean by word _all_ here?

File: src/server/api/routes/profile.router.js [hunk 2]
```diff
@@ -30,6 +31,40 @@ router.get('/student', (req, res, next) => {
...
+ *              type: string
+ *    responses:
+ *      200:
+ *        description: Profile was patched
+ *      5XX:
+ *        description: Unexpected error.
+ */
+router.patch('/profile', (req, res, next) => {
+  profileController
+    .editProfile(req.user.id)
```
Comment (tpetrina) [inline src/server/api/routes/profile.router.js]: Did you forget to pass body here?

**Review by tpetrina — CHANGES_REQUESTED** [review]
Looks good, but with one small issue. Have you tested it works?

File: src/server/api/routes/profile.router.js (hunk 2 above)
Comment (Elia-Git) [reply src/server/api/routes/profile.router.js]: Typo

Comment (Elia-Git) [reply src/server/api/routes/profile.router.js]: Fixed!

File: src/server/api/routes/profile.router.js (hunk 1 above)
Comment (Elia-Git) [reply src/server/api/routes/profile.router.js]: Typo!

**Review by tpetrina — APPROVED** [review]
Looks good!

**Review by pallabi-ghosh — APPROVED** [review]

**Pull request merged**